src/main/java/duke/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import duke.Storage;
+import duke.Ui;
+import duke.exceptions.DukeException;
+import duke.exceptions.MissingDateException;
+import duke.exceptions.UnknownInputException;
+import duke.tasks.Deadline;
+import duke.tasks.TaskList;
+
+public class AddDeadlineCommand extends Command {
+    private final Deadline deadline;
+
+    public AddDeadlineCommand(String description) throws DukeException {
+        if (description.isBlank()) {
+            throw new UnknownInputException("deadline");
+        }
+
+        String[] name = description.split(" /by ", 2);
```

Comment:
I noticed that earlier you used the variable name "description" for the String array after splitting, while here you also use the name "description" for a String type, I think it might be a bit confusing? Maybe labelling them as "descriptionArray" and "descriptionString" could make things clearer?

Reply:
i see.. i will take note of that